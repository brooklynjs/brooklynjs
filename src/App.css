a {
  border-bottom: 1px dotted;
  color: #DDD;
  text-decoration: none;
}

article {
  width: 500px;
}

small {
  color: #999;
  display: block;
  font-size: 16px;
}