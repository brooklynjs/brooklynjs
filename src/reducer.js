import { nextEventIndex } from './utils/events';

// Initial State
export const initialState = {
  allEvents: [],
  currentEvent: {
    eventDate: null,
    speakers: [],
    sponsors: []
  },
  currentEventIndex: null
};

// Actions
export const initialLoad = (events, now = new Date()) => ({
  type: 'INITIAL_LOAD', events, now
});

export const loadNextEvent = () => ({
  type: 'LOAD_NEXT_EVENT'
})

export const loadPreviousEvent = () => ({
  type: 'LOAD_PREVIOUS_EVENT'
});

// Reducer

const reducer = (state, action) => {
  switch(action.type) {
    case 'INITIAL_LOAD':
      return handleInitialLoad(state, action);

    case 'LOAD_NEXT_EVENT':
      return handleLoadNextEvent(state, action);

    case 'LOAD_PREVIOUS_EVENT':
      return handleLoadPreviousEvent(state, action);

    default:
      return state;
  }
};

const handleInitialLoad = (state, action) => {
  const { events, now } = action;
  const eventIndex = nextEventIndex(events, now);

  return {
    ...state,
    allEvents: events,
    currentEvent: events[eventIndex],
    currentEventIndex: eventIndex
  };
};

const handleLoadNextEvent = (state, action) => {
  const { allEvents, currentEventIndex } = state;

  if (currentEventIndex + 1 < allEvents.length) {
    const nextEventIndex = currentEventIndex + 1;
    const nextEvent = allEvents[nextEventIndex];

    return {
      ...state,
      currentEvent: nextEvent,
      currentEventIndex: nextEventIndex
    }
  } else {
    return state;
  }
};

const handleLoadPreviousEvent = (state, action) => {
  const { allEvents, currentEventIndex } = state;

  if (currentEventIndex > 0) {
    const previousEventIndex = currentEventIndex - 1;
    const previousEvent = allEvents[previousEventIndex];

    return {
      ...state,
      currentEvent: previousEvent,
      currentEventIndex: previousEventIndex
    };
  } else {
    return state;
  }
};

export default reducer;
