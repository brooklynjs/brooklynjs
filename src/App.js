import React from 'react';

import Footer from './components/Footer';
import Header from './components/Header';
import PlannedDisruption from './components/PlannedDisruption';
import Timeline from './components/Timeline';

import './App.css';

const state = {
  eventDate: new Date(2017, 8, 21),
  speakers: [],
  sponsors: []
};

const App = () => {
  const {
    eventDate,
    speakers,
    sponsors
  } = state;

  return (
    <div>
      <Header />

      <article>
        <PlannedDisruption eventDate={eventDate} />

        <Timeline
          eventDate={eventDate}
          speakers={speakers}
          sponsors={sponsors}
        />
      </article>

      <Footer />
    </div>
  );
}
export default App;
