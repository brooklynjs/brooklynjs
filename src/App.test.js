import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import App from './App';
import { initialState } from './reducer';

it('renders', () => {
  const tree = ReactTestRenderer.create(
    <App state={initialState} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
