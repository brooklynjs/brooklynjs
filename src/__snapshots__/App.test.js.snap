// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <header>
    Brooklyn
    <span
      className="j"
    >
      J
    </span>
    <span
      className="s"
    >
      S
    </span>
  </header>
  <article>
    <div
      className="planned-disruption"
    >
      <span>
        PLANNED DISRUPTION
      </span>
      <span />
      <br />
      <span>
        Please follow the directions below.
      </span>
    </div>
    <ol>
      <li
        className="end stop"
      >
        <time>
          now
        </time>
        <small>
          See the event board to
        </small>
        <div>
          <a
            href="http://tickets.brooklynjs.com"
          >
            Get your ticket
          </a>
           and 
          <a
            href="http://tickets.brooklynjs.com"
          >
            tell your friends
          </a>
        </div>
      </li>
      <li
        className="ride"
      />
      <li
        className="end stop"
      >
        <small>
          Check out our
        </small>
        <div>
          <a
            href="coc.html"
          >
            Code of Conduct
          </a>
        </div>
      </li>
      <li
        className="ride"
      />
      <li
        className="stop"
      >
        <time>
          
        </time>
        <small>
          Nothing to do but
        </small>
        <div>
          Wait patiently
        </div>
      </li>
      <li
        className="ride"
      />
      <li
        className="end stop"
      >
        <time />
        <small>
          Make your way to our awesome venue,
        </small>
        <div>
          <a
            href="//61local.com"
          >
            61 Local
          </a>
           on 
          <a
            href="//www.google.com/maps?q=61+Bergen+St+11201"
          >
            Bergen St.
          </a>
        </div>
      </li>
      <li
        className="ride"
      />
      <li
        className="end stop"
      >
        <small>
          Maybe come early and geek out for some
        </small>
        <div>
          <a
            href="http://tickets.brooklynjs.com"
          >
            Boroughgramming
          </a>
        </div>
      </li>
      <li
        className="walk"
      >
        <small />
        <div />
      </li>
      <li
        className="stop"
      >
        <time>
          7:00p
        </time>
        <small>
          Open the gray door in back, and
        </small>
        <div>
          Walk upstairs
        </div>
      </li>
      <li
        className="ride"
      />
      <li
        className="end stop"
      >
        <time>
          7:30p
        </time>
        <small>
          Grab a seat in time for
        </small>
        <div>
          Introduction and kickoff
        </div>
      </li>
      <div>
        <li
          className="ride"
        />
        <li
          className="speaker"
        >
          <time>
            8:00p
          </time>
          <small>
            Listen to our 
            first
             speaker, or better yet
          </small>
          <div
            className="empty-speaker"
          >
            <a
              href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new"
            >
              Be our 
              first
               speaker
            </a>
          </div>
        </li>
      </div>
      <div>
        <li
          className="ride"
        />
        <li
          className="speaker"
        >
          <time>
            8:15p
          </time>
          <small>
            Listen to our 
            second
             speaker, or better yet
          </small>
          <div
            className="empty-speaker"
          >
            <a
              href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new"
            >
              Be our 
              second
               speaker
            </a>
          </div>
        </li>
      </div>
      <div>
        <li
          className="ride"
        />
        <li
          className="speaker"
        >
          <time>
            8:30p
          </time>
          <small>
            Listen to our 
            third
             speaker, or better yet
          </small>
          <div
            className="empty-speaker"
          >
            <a
              href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new"
            >
              Be our 
              third
               speaker
            </a>
          </div>
        </li>
      </div>
      <div>
        <li
          className="ride"
        />
        <li
          className="speaker"
        >
          <time>
            8:45p
          </time>
          <small>
            Listen to our 
            fourth
             speaker, or better yet
          </small>
          <div
            className="empty-speaker"
          >
            <a
              href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new"
            >
              Be our 
              fourth
               speaker
            </a>
          </div>
        </li>
      </div>
      <div>
        <li
          className="ride"
        />
        <li
          className="speaker"
        >
          <time>
            9:00p
          </time>
          <small>
            Listen to our 
            fifth
             speaker, or better yet
          </small>
          <div
            className="empty-speaker"
          >
            <a
              href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new"
            >
              Be our 
              fifth
               speaker
            </a>
          </div>
        </li>
      </div>
      <li
        className="ride"
      />
      <li
        className="stop"
      >
        <time>
          9:15p
        </time>
        <small>
          Wind down with the
        </small>
        <div>
          Wrap-up
        </div>
      </li>
      <li
        className="walk"
      >
        <small>
          Round-trip downstairs to the bar to get some
        </small>
        <div>
          Local 
          <a
            href="//www.beermenus.com/places/2237-61-local"
          >
            craft beer
          </a>
           / soda
        </div>
      </li>
      <li
        className="stop"
      >
        <time>
          12:00a
        </time>
        <small>
          Geek out with your boroughmates during
        </small>
        <div>
          The bevera.js social hour(s)
        </div>
      </li>
      <li
        className="ride"
      />
      <li
        className="end stop sponsors"
      >
        <small>
          This event made possible by our sponsors
        </small>
        <div>
          <a
            href="sponsorship.html"
          >
            Be our next sponsor.
          </a>
        </div>
      </li>
    </ol>
  </article>
  <footer>
    BrooklynJS expects all speakers and attendees
    <br />
    to follow the 
    <a
      href="//jsconf.com/codeofconduct.html"
    >
      JSConf code of conduct
    </a>
    . For more information about us, see our FAQs for 
    <a
      href="https://github.com/brooklynjs/brooklynjs.github.io/blob/master/speakers.md"
    >
      speakers
    </a>
    , 
    <a
      href="https://github.com/brooklynjs/brooklynjs.github.io/blob/master/attendee-faq.md"
    >
      attendees
    </a>
    , and 
    <a
      href="sponsorship.html"
    >
      sponsors
    </a>
    , or 
    <a
      href="https://github.com/jed/building-brooklynjs"
    >
      learn more about our history
    </a>
    .
  </footer>
</div>
`;
